src/zenml/config/server_config.py

```diff
@@ -144,6 +149,8 @@ class ServerConfiguration(BaseModel):
     external_cookie_name: Optional[str] = None
     external_server_id: Optional[UUID] = None
 
+    metadata: Optional[Dict[str, Any]] = None
```

Comment:
You need to add some code to the root validator below to convert the metadata from string/JSON format into dictionary. Something like this:

```
        if isinstance(values.get("metadata"), str):
            try:
                values["metadata"] = json.loads(values["metadata"])
            except json.JSONDecodeError as e:
                raise ValueError(
                    f"The server metadata is not a valid JSON object: {e}"
                )
```

For an example of this, please see here: https://github.com/zenml-io/zenml/blob/develop/src/zenml/zen_stores/secrets_stores/service_connector_secrets_store.py#L72-L80

Reply:
ah yes, good catch! I had the impression that I had to load it into json somewhere but then this thought got lost with all the code re-org between the repos